src/js/hooks/cycleCount/useCountStep.jsx

```diff
@@ -302,15 +303,19 @@ const useCountStep = () => {
       show();
       resetValidationState();
       for (const cycleCount of tableData.current) {
-        const cycleCountItemsToUpdate = cycleCount.cycleCountItems.filter((item) => (item.updated && !item.id.includes('newRow')));
+        const cycleCountItemsToUpdate = cycleCount.cycleCountItems
+          .filter((item) => (item.updated && !item.id.includes('newRow')))
+          .map(trimLotNumberSpaces);
```

Comment:
so you're also trimming the lot number when updating non-custom rows? What happens if we have a pre-existing item in our system with lot `"ABC   "` (some whitespace on the end). Won't updating the countItem now create a new inventory item for the lot `"ABC"` (no spaces)?

It might seem like an edge case but we actually have 1261 rows like that:
```
SELECT id, lot_number FROM inventory_item WHERE lot_number LIKE "% ";
```

Long term ideally we can migrate our existing data and stop allowing trailing whitespace for lot, but maybe the simplest short term solution is to only trim the lot number on items to update if the lot number has changed, otherwise leave it as is

Reply:
> What happens if we have a pre-existing item in our system with lot "ABC   " (some whitespace on the end). Won't updating the countItem now create a new inventory item for the lot "ABC" (no spaces)?

We discussed this case on our internal standup in the morning with Manon and we confirmed this would eventually create a new inventory item, but it would be expected, as the "dirty data" should eventually be removed by a user.

Reply:
cool as long as it's known and was discussed :+1: